scalahost/nsc/src/main/scala/scala/meta/internal/scalahost/ScalahostPipeline.scala

```diff
@@ -15,6 +15,20 @@ import scala.meta.internal.semantic.{vfs => v}
 import scala.meta.internal.semantic.{schema => s}
 
 trait ScalahostPipeline extends DatabaseOps { self: ScalahostPlugin =>
+  // NOTE: Here we encode assumptions that hold by design:
+  //   * Output directory stores the semantic db generated by the previous compilation
+  //     (that is necessary for incremental compilation support)
+  //   * Working directory represents both previous and current compilation roots
+  //     (that's the protocol that we assume that Scala build tools support)
+  //   * All uris related to the previous semantic db are based on the file: protocol
+  //     (that follows from the fact that the output directory is based on the file: protocol)
+  lazy val scalametaClasspath = Classpath(
+    global.settings.outputDirs.getSingleOutput
+      .map(_.file.getAbsolutePath)
+      .getOrElse(global.settings.d.value))
+  lazy val scalametaSourcepath = Sourcepath(
+    sys.props.getOrElse("scalameta.sourcepath", sys.props("user.dir")))
```

Comment:
Configuration 👏 How about we move this to `ConfigOps` so there's one place to look these things up?